### PR TITLE
Make Openstack "trust-device-path" configurable via datacenters.yaml

### LINKED ISSUE #3110: Make CloudConfig parameter `trust-device-path` configurable for OpenStack datacenters
**User Story**
Some OpenStack datacenters require that the Kubelet uses the `trust-device-path` flag inside the CloudConfig.
To be able to configure this with Kubermatic, the customer would need to configure this setting inside the datacenters.yaml, like with https://github.com/kubermatic/kubermatic/blob/master/api/pkg/provider/datacenter.go#L48.

As this flag needs to be passed through to the Kubelet, we should use the `OverwriteCloudConfig` flag inside the Machine object ([Example](https://github.com/kubermatic/kubermatic/blob/master/api/pkg/resources/machine/machinedeployment.go#L98)).
It lets us supply a custom cloud config to the machine, so the machine-controller will not create a custom one.

**Acceptance criteria**
- [ ] The cloud config on the worker node uses the `trust-device-path` based on a flag from the Datacenters.yaml
- [ ] A entry in the docs has been created to document this new override setting

### PR DESCRIPTION
**Pull request by nikhita**
**What this PR does / why we need it**:

This makes the `trust-device-path` setting on Openstack datacenters configurable.

Related change to machine controller: https://github.com/kubermatic/machine-controller/pull/506
Machine-controller bump: https://github.com/kubermatic/kubermatic/pull/3298

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubermatic/kubermatic/issues/3110

**Does this PR introduce a user-facing change?**:

```release-note
The trust-device-path cloud config property of Openstack clusters can be configured via datacenters.yaml.
```

### CODE REVIEW

File: api/pkg/resources/machine/machine.go [hunk 1]
```diff
@@ -67,6 +67,16 @@ func Machine(c *kubermaticv1.Cluster, node *apiv1.Node, dc provider.DatacenterMe
...
 		config.CloudProvider = providerconfig.CloudProviderOpenstack
+
+		// We use OverwriteCloudConfig for Openstack to ensure we always
+		// use the credentials passed in via frontend for the cloud-provider
+		// functionality
+		overwriteCloudConfig, err := cloudconfig.CloudConfig(c, &dc)
+		if err != nil {
+			return nil, err
+		}
+		config.OverwriteCloudConfig = &overwriteCloudConfig
```
Comment (mrIncompetent) [inline api/pkg/resources/machine/machine.go]: See https://github.com/kubermatic/kubermatic/issues/3110#issuecomment-480181313

**Comment by nikhita** [comment]
Updated, but it needs https://github.com/kubermatic/kubermatic/pull/3298 to merge first.

**Comment by nikhita** [comment]
/retest
/hold cancel 
https://github.com/kubermatic/kubermatic/pull/3298 has merged

**Pull request merged**